components/cache/index.js

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const async = require('async');
+const redis = require('redis');
+const transactionTypes = require('../../helpers/transaction_types');
+const { CACHE } = require('./constants');
+
+const errorCacheDisabled = 'Cache Disabled';
+
+/**
+ * Cache module.
+ *
+ * @class
+ * @memberof components
+ * @see Parent: {@link components}
+ * @requires async
+ * @requires redis
+ * @requires helpers/transaction_types
+ * @param {Object} options - Cache options
+ * @param {Object} logger
+ * @param {function} cb
+ * @todo Add description for the function and the params
+ */
+class Cache {
+	constructor(options, logger) {
+		this.options = options;
+		this.logger = logger;
+		this.cacheReady = false;
+	}
+
+	connect(cb) {
+		this.client = redis.createClient(this.options);
+		this.client.once('error', this._onConnectionError.bind(this, () => cb()));
+		this.client.once('ready', this._onReady.bind(this, () => cb()));
+	}
```

Comment:
As we are refactoring, we are trying to avoid callback patterns. So suggested to return Promise from this function.